nautobot/core/api/metadata.py

```diff
@@ -72,41 +77,47 @@ def _get_field_properties(self, field: schema.SerializerType, name: str) -> Dict
             result["items"] = self._get_field_properties(field.child_relation, "")
             result["uniqueItems"] = True
         else:
+            if isinstance(field, drf_serializers.RelatedField):
+                result["uniqueItems"] = True
+                if hasattr(field, "queryset") and hasattr(field.queryset, "model"):
+                    model_options = field.queryset.model._meta
+                    result["required"] = field.required
+                    # Custom Keyword: modelName and appLabel
+                    # This Keyword represents the model name of the uuid model
+                    # and appLabel represents the app_name of the model
+                    result["modelName"] = model_options.model_name
+                    result["appLabel"] = model_options.app_label
             if field.allow_null:
                 result["type"] = [result["type"], "null"]
-            enum = type_map_obj.get("enum")
-            if enum:
+            if enum := type_map_obj.get("enum"):
                 if enum == "choices":
                     choices = field.choices
                     result["enum"] = list(choices.keys())
                     result["enumNames"] = list(choices.values())
                 if isinstance(enum, (list, tuple)):
-                    if isinstance(enum, (list, tuple)):
-                        result["enum"] = [item[0] for item in enum]
-                        result["enumNames"] = [item[1] for item in enum]
-                    else:
-                        result["enum"] = enum
-                        result["enumNames"] = list(enum)
-
-            # Process "format"
-            format_ = type_map_obj["format"]
-            if format_:
+                    result["enum"] = [item[0] for item in enum]
+                    result["enumNames"] = [item[1] for item in enum]
+            if format_ := type_map_obj["format"]:
                 result["format"] = format_
 
-            # Process "readOnly"
-            read_only = type_map_obj["readOnly"]
-            if read_only:
+            if read_only := field.read_only:
```

Comment:
Why the change from `type_map_obj["readOnly"]` to `field.read_only`?

Reply:
Its not always accurate: below response of `print("====> " + name, (type_map_obj["readOnly"], field.read_only))`

```bash
====> id (None, True)
====> object_type (None, True)
====> display (None, True)
====> url (None, True)
====> natural_key_slug (None, True)
====> tree_depth (None, True)
====> time_zone (None, False)
====> circuit_count (None, True)
====> device_count (None, True)
====> name (None, False)
====> slug (None, False)
====> description (None, False)
====> facility (None, False)
====> asn (None, False)
```

Reply:
I had added this and higher up being processed by `_get_type_map_value()`, which was relying on it being set in the schema somehow, which explains why it wasn't always accurate.